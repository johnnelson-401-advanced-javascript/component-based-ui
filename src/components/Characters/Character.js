import React from 'react';
import PropTypes from 'prop-types';

export default function Character({ image, name, species }) {
  return (
    <dl>
      <img src={image}></img>

      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Species</dt>
      <dd>{species}</dd>
    </dl>
  );
}

Character.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string.isRequired,
  species: PropTypes.string
};
