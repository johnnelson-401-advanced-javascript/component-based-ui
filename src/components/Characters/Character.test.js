import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character.js', () => {
  it('renders a Character', () => {
    const character = {
      'name': 'Noob-Noob',
      'status': 'Alive',
      'species': 'Poopybutthole',
      'image': 'https://rickandmortyapi.com/api/character/avatar/252.jpeg'
    };
    let wrapper = shallow(<Character 
      name={character.name}
      image={character.image}
      species={character.species}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
