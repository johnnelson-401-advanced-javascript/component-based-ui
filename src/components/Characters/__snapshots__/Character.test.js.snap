// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character.js renders a Character 1`] = `
<dl
  className="characterDisplay"
>
  <img
    src="https://rickandmortyapi.com/api/character/avatar/252.jpeg"
  />
  <dt>
    Name
  </dt>
  <dd>
    Noob-Noob
  </dd>
  <dt>
    Species
  </dt>
  <dd>
    Poopybutthole
  </dd>
</dl>
`;
